File: El Mamoun/Lv 1/Problem 28.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
/*
 Problem 28: Write a program to Sum odd numbers from 1 to N.

 */
namespace Problem_28
{
    internal class Program
    {
        enum enEvenOrOdd { Even = 1, Odd = 2 };
        static int ReadNumber()
        {
            Console.WriteLine("Please Enter a Number?:");
            int Number = Convert.ToInt32(Console.ReadLine());

            return Number;
        }
        static enEvenOrOdd CheckEvenOrOdd(int Number)
        {
            if (Number % 2 == 0)
                return enEvenOrOdd.Even;

            else
                return enEvenOrOdd.Odd;
        }
        static int SumOddNumber(int Number)
        {
            int SumOddNumber = 0;
            
            for (int Counter = 1; Counter < Number; Counter++)
            {
                if (CheckEvenOrOdd(Counter) == enEvenOrOdd.Odd)
                    SumOddNumber += Counter;
            }
            return SumOddNumber;
        }
        static void PrintOddNumberFrom1ToN(int SumOddNumbr)
        {
            Console.WriteLine("Sum Odd Number is: " + SumOddNumbr);
        }

        static void Main(string[] args)
        {

            PrintOddNumberFrom1ToN(SumOddNumber(ReadNumber()));

            Console.ReadKey();
        }
    }
}

File: El Mamoun/Lv 1/Problem 28.cpp
#include<iostream>
using namespace std;

/*
 Problem 28: Write a program to Sum odd numbers from 1 to N.
 */

enum enEvenOrOdd { Even = 1, Odd = 2 };

enEvenOrOdd CheckEvenOrOdd(int number)
{

	if (number % 2 != 0)
		return enEvenOrOdd::Odd;

	else
		return enEvenOrOdd::Even;
}

int ReadNumber()
{
	int number = 0;

	cout << "Please Enter a Number? ";
	cin >> number;

	return number;
}

int SumOddNumberFrom1ToN(int Number)
{
	int SumOddNumbers = 0;
	
	cout << "\nOdd Number = ";
	for (int Counter = 1; Counter <= Number; Counter++)
	{

		if (CheckEvenOrOdd(Counter) == enEvenOrOdd::Odd)
			SumOddNumbers += Counter;
	}
	return SumOddNumbers;
}

void PrintOddNumberFrom1ToN(int SumOddNumber)
{
	cout << "\nSum Odd Number is: " << SumOddNumber;
}




int main()
{

	cout << SumOddNumberFrom1ToN(ReadNumber());
	system("Pause > 0");

	return 0;
}